modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/listk.kt

```diff
@@ -321,6 +323,44 @@ interface ListKSemialign : Semialign<ForListK>, ListKFunctor {
   }
 }
 
+/**
+ * Left-padded zipWith.
+ */
+fun <A, B, C> ListK<A>.lpadZipWith(
```

Comment:
I understand why this has to be in this module, I am just not sure if this is the right file for this. I am also not aware of any other methods like this on other datatypes. @pakoito Is this in the right place?

Reply:
I'd put it in the original ListK file tbh, unless there's an import issue

Reply:
Well yes, it uses methods from the semialign instance. I think the current workaround is to specialise all methods needed to list and implement the instance using those. This means moving the implementation of all of those methods to `ListK`. So `padZipWith`, `padZip` and `alignWith` and while at it probably also `align` itself.

Reply:
ok. i have moved the functions + align to ListK